core/src/kura.rs

```diff
@@ -362,15 +355,30 @@ impl Unlocked {
 
     /// Try to acquire lockfile and convert self to [`Locked`]
     fn lock(self) -> Result<Locked> {
+        let path = self.join(LOCK_FILE_NAME);
         if let Err(e) = fs::File::options()
             .read(true)
             .write(true)
             .create_new(true)
-            .open(self.join(LOCK_FILE_NAME))
+            .open(path.clone())
         {
             match e.kind() {
                 std::io::ErrorKind::AlreadyExists => Err(Error::Locked(self.0)),
-                _ => Err(e.into()),
+                std::io::ErrorKind::NotFound => {
+                    std::fs::create_dir_all(self.0.clone())
```

Comment:
Can't we just call this pre-emptively?

Reply:
It doesn't give us much and falls in line with the "apologise if wrong, instead of asking" paradigm.